### ISSUE TITLE
Memory Leak

### ISSUE THREAD

**Issue by Glussian** [cbpowell/MarqueeLabel#133]
There are memory leaks on both swift and Objective C.
To reproduce, run instruments/Memory leaks on the project demo.

**Comment by CodeEagle** [comment]
here we go 
![image](https://cloud.githubusercontent.com/assets/2311352/14700410/f11bbb30-07d1-11e6-9a50-a1ee25789d4f.png)

**Comment by cbpowell** [comment]
Thanks! I checked it out as well. It seems to leak when a scroll animation is cancelled. Starting & quickly cancelling an animation happens several times during loading of the view controller, so the same leaks stack up it seems. I was able to get to zero leaks by changing the default value of `holdScrolling` to true, which heads off those initial start/cancels. Manually turning `holdScrolling` off then starts the animation, which loops without any leaks.

The primary leaked object seems to be the completion block on those cancelled animations? I'm still looking into it, I'm not sure why the completion block is being leaked when the animation it's attached to is cancelled (via `removeAllAnimations()`). Any thoughts?

**Comment by cbpowell** [comment]
Well I never figured out _why_ the blocks were leaking, but I worked around it with a probably-safer method. With 9272a78862e215e7bfe788acdf79d812d44e29ca and 8b33189294462cbe33e3e4fea905d56c99275b0d the completion blocks are stored in a private property of MarqueeLabel and then fired upon animation completion. This seems to eliminate all the memory leaks for me! I also took the opportunity to reduce some potential strong references to self in the various closures.
